EventFilter/Utilities/src/DirManager.cc

```diff
@@ -59,13 +60,12 @@ namespace evf {
     retval += tmpdir;
     return retval;
   }
+
   bool DirManager::checkDirEmpty(std::string &d) {
-    int filecount = 0;
     DIR *dir = opendir(d.c_str());
-    struct dirent *buf;
-    while ((buf = readdir(dir))) {
-      filecount++;
+    while (readdir(dir)) {
+      return false;
```

Comment:
why is there a ```while``` still needed if the first entry would break out of the loop?
this suggests a change in behavior, if ```while(readdir``` was meant to do some repeated action.

Reply:
I think this function was already badly formed, as it misses a final call to `closedir(dir)` which would reset the directory stream.
On the other hand, I agree that stopping reading the dir trees as soon as it has been checked that they are not empty (which is the purpose of the function) may let the internal pointer associated to a different position in the directory stream: therefore I agree with you and I'm going to restore the exact previous behaviour, just getting rid of the nested dead assignment. Luckily enough, this function does not seem to be used anywhere in CMSSW, thus we don't have to care (for now) of the effect of a missing `closedir(dir)` statement. I'd let @cms-sw/daq-l2 (whose signature is needed anyhow here) to judge whether the `DirManager::checkDirEmpty(...)` has to be removed, fixed, or it is correct to keep it as it is now.